mantidproject/mslice
# 942 increase all fonts in plot

**Summary of Changes**
- Added widget to set all fonts in plot to a given value, present when Plot Options is opened.
- Added another widget to control scaling of all fonts

**Description of work:**
- Three main components were altered: the GUI window (also called view) class, the cut and slice plots classes, and the PlotOptionsPresenter class (which functions and the link between the GUI and the internal class for controlling plot options)
- The GUI window: Added four widgets in total: two checkboxes and two line edits:
![image](https://github.com/mantidproject/mslice/assets/80104863/8b3170da-f459-4c22-8304-3ffa48ebf9f6)
- Added functions in Plot Options class that controls how the checkboxes interact with the line edits and between themselves (can't check the two checkboxes simultaneously, and can only edit line edits if checkbox is checked)
- Added signals in Plot Options class that are caught by PlotOptionsPresenter in `plot_options_presenter.py` to signal whether the two new fields (for setting all font sizes and scaling all font sizes) are part of the modified values. Added a new function also in this class that removes elements from the dictionary of modified values. This function is triggered when user unticks either of the checkboxes, so that any changes previously made are internally discarded from the dict of modified values. Similarly, when a checkbox is ticked, the default values are loaded onto`modified values` and any sunbsequent edit by the user in the Gui apdatest these values.
- In `cut_plot.py`, the implementation of setting and scaling all fonts is contained whithin the funcitons `CutPlot.all_fonts_size` and `CutPlot.scale_all_fonts` respectively. These two functions are implemented by getter and setter decorators.
- Added tests for cut and slice plots, on the implementation of setting and scaling all fonts. Added another test in `cut_options_presenter_test.py` to confirm that data transfer between GUI view and internal class is happening successfully.
- Manually confirmed that all functunality is preserved whether the plot options window is kept open or not.

**To test:**

- Open plot options window and play around with the new widgets. 
    - Change fonts of plot one by one to different sizes and check that setting all fonts to the same size works.
    - Change fonts to different sizes again and check that scaling all fonts preserves proportions.
    - Enable and disable checkboxes and confirm that plot fonts remain unaltered.
    - Check that it is not possible to simultaneously setect both checkboxes.
    - Enable one checkbox but don't edit the corresponding line, changes should still be applied to plot
 - Change one of the values in the tests for slice and cut plots and check that test fails.
 - In `cut_options_presenter_test.py`, comment out any line that calls `self.presenter._value_modified` and check that test fails (this is the equivalent of no checkboxes being ticked)

Fixes #942 .
